frontend/src/api/apiMergeUtils.ts

```diff
@@ -1,10 +1,33 @@
-import { QueryParams } from '@openshift/dynamic-plugin-sdk-utils';
+import { K8sResourceBaseOptions, QueryParams } from '@openshift/dynamic-plugin-sdk-utils';
 import { K8sAPIOptions } from '~/k8sTypes';
 
-export const mergeK8sQueryParams = (
+const mergeK8sQueryParams = (
   opts: K8sAPIOptions = {},
   specificOpts: QueryParams = {},
 ): QueryParams => ({
   ...specificOpts,
   ...(opts.dryRun && { dryRun: 'All' }),
 });
+
+const mergeRequestInit = (
+  opts: K8sAPIOptions = {},
+  specificOpts: RequestInit = {},
+): RequestInit => ({
+  ...specificOpts,
+  ...(opts.signal && { signal: opts.signal }),
+});
+
+export const applyK8sAPIOptions = <T extends K8sResourceBaseOptions>(
+  opts: K8sAPIOptions = {},
+  apiData: T,
+): T => ({
+  ...apiData,
+  queryOptions: {
+    ...apiData.queryOptions,
+    queryParams: mergeK8sQueryParams(opts, apiData.queryOptions?.queryParams),
+  },
+  fetchOptions: {
+    ...apiData.fetchOptions,
+    requestInit: mergeRequestInit(opts, apiData.fetchOptions?.requestInit),
```

Comment:
should there be another `?` here 
```tsx
apiData?.fetchOptions?.requestInit
```

Reply:
in that case above, only queryOptions is provided, not a fetchOptions. which would cause an error here `apiData.fetchOptions?.requestInit` right?

I may be wrong and that doesnt cause an error

Reply:
An empty `requestInit` won't cause an issue. The `mergeRequestInit` handles optionals and defaults them to `{}`.